### PR TITLE
fix(openapi): fix path item overriding problem

### LINKED ISSUE #137: Path item overriding problem in OpenAPI
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind feature

/kind bug

**What happened**:
Can't build a nested descriptor

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
I'll fix today(02.05)

A PR related with it #134

### PR DESCRIPTION
**Pull request by liubog2008**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #137 

**Special notes for your reviewer**:

/cc @ddysher 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/openapi-gen/builder/openapi.go [hunk 1]
```diff
@@ -99,17 +102,18 @@ func (o *openAPI) buildDescriptor(path string, consumes, produces []string, desc
...
 		}
 	}
 	return nil
 }
 
-func (o *openAPI) buildPathItems(path string, consumes, produces []string, defs []definition.Definition) (map[string]spec.PathItem, error) {
-	specPaths := map[string]spec.PathItem{}
+func (o *openAPI) buildPathItem(consumes, produces []string, defs []definition.Definition) (*spec.PathItem, error) {
+	pathItem := spec.PathItem{}
+	hasDef := false
```
Comment (kdada) [inline cmd/openapi-gen/builder/openapi.go]: Why not just check the length of `defs`?

File: cmd/openapi-gen/builder/openapi.go [hunk 2]
```diff
@@ -99,17 +102,18 @@ func (o *openAPI) buildDescriptor(path string, consumes, produces []string, desc
...
 	for _, def := range defs {
-		pathItem := spec.PathItem{}
+		hasDef = true
 		op, err := o.buildOperation(&def, consumes, produces, service.HTTPCodeFor(def.Method))
```
Comment (kdada) [inline cmd/openapi-gen/builder/openapi.go]: Does it support **two or more** definitions with same method but have different consume-produce peer?

e.g.
```

definition.Definition {
    Method: Create,
    Consumes: []string{JSON},
    Produces: []string{JSON},
},
definition.Definition {
    Method: Create,
    Consumes: []string{File},
    Produces: []string{JSON},
},
```

Comment (liubog2008) [reply cmd/openapi-gen/builder/openapi.go]: 。。。

File: cmd/openapi-gen/builder/openapi_test.go [hunk 3]
```diff
@@ -299,21 +299,19 @@ func newConfig() *common.Config {
 	}
 }
 
-func TestBuildPathItems(t *testing.T) {
+func TestBuildPathItem(t *testing.T) {
```
Comment (ddysher) [inline cmd/openapi-gen/builder/openapi_test.go]: Add a new test for nested descriptor?

File: cmd/openapi-gen/builder/openapi.go (hunk 2 above)
Comment (ddysher) [reply cmd/openapi-gen/builder/openapi.go]: what does `。。。`  mean?  not supported?

**Comment by ddysher** [comment]
@liubog2008 ping

**Comment by liubog2008** [comment]
@ddysher PTAL

**Comment by ddysher** [comment]
@liubog2008 can u update ur mail address?

**Pull request merged**